src/archivematicaCommon/lib/archivematicaFunctions.py

```diff
@@ -200,3 +200,21 @@ def create_structured_directory(basepath, manual_normalization=False, printing=F
     create_directories(REQUIRED_DIRECTORIES, basepath=basepath, printing=printing)
     if manual_normalization:
         create_directories(MANUAL_NORMALIZATION_DIRECTORIES, basepath=basepath, printing=printing)
+
+
+def close_db_connections(fn):
+    """
+    Decorator to wrap a threaded function in a try-catch ensuring that the
+    database connections are closed if unrecoverable errors have occurred or if
+    it outlived its maximum age.
+
+    Similar to Django's behavior within HTTP request cycles.
+    """
+    def wrapped(*args,  **kwargs):
+        from django.db import close_old_connections
+        try:
+            close_old_connections()
```

Comment:
I was trying to follow [Django's cleaning process before and after HTTP requests](https://github.com/django/django/blob/master/django/db/__init__.py#L60-L66) to the letter, but on a second thought I think you are totally right as we are not reusing threads in the same way that uWSGI is reusing processes.
I've tried without the first call and it seems to work.